daffodil-test/src/test/resources/org/apache/daffodil/section23/dfdl_expressions/expressions.tdml

```diff
@@ -2309,23 +2309,50 @@
   <tdml:defineSchema name="checkConstraints">
     <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
     <dfdl:format ref="ex:GeneralFormat" />
+
+    <xs:simpleType name="cnt1">
+      <xs:restriction base="xs:int">
+        <xs:minInclusive value="0"/>
+        <xs:maxInclusive value="2"/>
+      </xs:restriction>
+    </xs:simpleType>
+
     <xs:element name="e1">
       <xs:complexType>
         <xs:sequence>
-          <xs:element name="cnt" type="xs:int"
-            dfdl:lengthKind="delimited" />
-          <xs:element name="a" minOccurs="0" maxOccurs="unbounded"
-            dfdl:occursCountKind="expression" dfdl:occursCount="{ xs:int(../ex:cnt) }">
-            <xs:complexType>
+          <xs:element name="cnt" type="xs:int" dfdl:lengthKind="delimited" />
               <xs:sequence>
-                <xs:element name="v" type="xs:int"
-                  dfdl:inputValueCalc="{ dfdl:checkConstraints(.) }" />
+
+                <!-- These Pass -->
+
+<!--                <xs:element name="v" type="ex:cnt1"-->
+<!--                            dfdl:inputValueCalc="{ if (dfdl:checkConstraints(../ex:cnt)) then xs:int(../ex:cnt) else -1 }" />-->
+
+<!--                <xs:element name="v" type="ex:cnt1"-->
+<!--                            dfdl:inputValueCalc="{ if (dfdl:checkConstraints(./../ex:cnt)) then xs:int(../ex:cnt) else -1 }" />-->
+
+<!--                <xs:element name="v" type="ex:cnt1"-->
+<!--                            dfdl:inputValueCalc="{ if (dfdl:checkConstraints(./../ex:cnt/.)) then xs:int(../ex:cnt) else -1 }" />-->
+
+                <!-- These Fail -->
+
+<!--                <xs:element name="v" type="ex:cnt1"-->
+<!--                  dfdl:inputValueCalc="{ if (dfdl:checkConstraints(.)) then xs:int(../ex:cnt) else -1 }" />-->
+
+<!--                <xs:element name="v" type="xs:int" dfdl:lengthKind="explicit" dfdl:length="{.}" />-->
+
+                <!-- Pass -->
+                    <xs:element name="v" type="ex:cnt1"
+                            dfdl:inputValueCalc="{ dfdl:checkConstraints(if(../ex:cnt eq 1) then ../ex:cnt else .) }" />
+                <!-- Fail -->
+<!--                    <xs:element name="v" type="ex:cnt1"-->
+<!--                            dfdl:inputValueCalc="{ dfdl:checkConstraints(if(../ex:cnt eq 2) then ../ex:cnt else .) }" />-->
```

Comment:
Rather than having comments for certain things that pass fail, I woulc recommend creating a unique test for each of these. That ensures that we do not have any regressions for the expected behavior.

Reply:
Will do, just wanted to make sure I was on the right track before building them out.